### ISSUE TITLE
Custom entrytypes: inheritance of valid field types 

### ISSUE THREAD

**Issue by schoeps** [plk/biber#300]
I am using  Biber 2.14 in tool mode. When declaring a new datamodel configuration in a user configuration file (see `biber.conf` below) then a new entrytype (`xyz`) does not inherit valid field types (e.g. `FILE`) from the main biber-tool.conf file but vice-versa: existing entries see new fieldtypes:
```
biber --tool --validate-datamodel data.bib
> WARN - Datamodel: Entry 'doe' (data.bib): Invalid field 'file' for entrytype 'xyz'
```
I guess this is a bug and not a design decision?

data.bib
```
@xyz{doe,
	author = {Doe, J.},
	title  = {Someting important},
	file   = {doe.pdf},
}
@bibnote{miller,
	abc = {some info},
}
```

biber.conf
```
<?xml version="1.0" encoding="UTF-8"?>
<config>
	<datamodel>
		<entrytypes>
			<entrytype>xyz</entrytype>
		</entrytypes>
		<entryfields>
			<entrytype>xyz</entrytype>
			<field>author</field>
			<field>title</field>
		</entryfields>
		<entryfields>
			<field>abc</field>
		</entryfields>
		<fields>
			<field fieldtype="field" datatype="literal">abc</field>
		</fields>
	</datamodel>
</config>
```

**Comment by plk** [comment]
It is a bug, thank you for reporting it. I am going to overhaul the datamodel merging code anyway and will update here when done. Should be this weekend.

**Comment by plk** [comment]
Biber DEV 2.15 should fix this which is now on Sourceforge. The datamodel customisation has been improved so that additions/modifications through the .conf file should now all work properly. If you with to delete something from the default datamodel or heavily modify the default datamode, use the new `--no-default-datamodel` option and supply a complete datamodel.

**Comment by schoeps** [comment]
Thanks! This bug of the MWE is indeed fixed. However, I will probably define my own datamodel and use `--no-default-datamodel`. I have defined custom inheritance rules and they still mess up parsing.

**Background** (only read if you care...): I created some custom types `event`, `poster`, `talk` and `journal`.  `poster`, `talk` shall inherit via crossref from `event` (e.g. `title` will be inherited as `eventtitle`,`location`,...) while articles can inherit from `journal` (e.g. `title` shall be inherited as `journaltitle`, `issn`, `shorttitle`,....).
